src/request.ts

```diff
@@ -483,12 +490,14 @@ class DatastoreRequest {
    * @param {?error} callback.err An error returned while making this request
    * @param {object} callback.apiResponse The full API response.
    */
-  insert(entities, callback) {
+  insert(entities: Entities): Promise<google.datastore.v1.ICommitResponse>;
+  insert(entities: Entities, callback: CallOptions): void;
+  insert(entities: Entities, callback?: CallOptions): (void|Promise<google.datastore.v1.ICommitResponse>) {
```

Comment:
This is good, but you don't need the wrapping parens in `(void|Promise<google.datastore.v1.ICommitResponse>)`

Reply:
Thanks, okay!